Unit tests fail under python2.7

Unit tests are failing under python2.7:

======================================================================
FAIL: test_root_static_path (tornado.test.web_test.StaticFileTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tornado/testing.py", line 132, in __call__
    result = self.orig_method(*args, **kwargs)
  File "tornado/test/web_test.py", line 1223, in test_root_static_path
    self.assertEqual(response.code, 200)
AssertionError: 404 != 200

======================================================================
FAIL: test_root_static_path (tornado.test.wsgi_test.WSGIAdapterWrappedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tornado/testing.py", line 132, in __call__
    result = self.orig_method(*args, **kwargs)
  File "tornado/test/web_test.py", line 1223, in test_root_static_path
    self.assertEqual(response.code, 200)
AssertionError: 404 != 200

======================================================================
FAIL: test_root_static_path (tornado.test.wsgi_test.WSGIApplicationWrappedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tornado/testing.py", line 132, in __call__
    result = self.orig_method(*args, **kwargs)
  File "tornado/test/web_test.py", line 1223, in test_root_static_path
    self.assertEqual(response.code, 200)
AssertionError: 404 != 200

--- a/tornado/test/web_test.py
+++ b/tornado/test/web_test.py
@@ -1218,8 +1218,9 @@ class StaticFileTest(WebTestCase):
         # to disable Tornado's path validation (in conjunction with
         # their own validation in get_absolute_path). Make sure
         # that the stricter validation in 4.2.1 doesn't break them.
-        path = os.path.join(os.path.dirname(__file__), 'static/robots.txt')
-        response = self.get_and_head('/root_static' + path)
+        path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                            'static/robots.txt')
+        response = self.get_and_head('/root_static' + urllib_parse.quote(path))
         self.assertEqual(response.code, 200)
 
 

